edx_dl/edx_dl.py

```diff
@@ -598,24 +599,24 @@ def main():
     # notice that we could iterate over all_units, but we prefer to do it over
     # sections/subsections to add correct prefixes and shows nicer information
     video_format_option = args.format + '/mp4' if args.format else 'mp4'
-    subtitles_option = '--all-subs' if args.subtitles else ''
     coursename = directory_name(selected_course.name)
     for selected_section in selected_sections:
-        section_dirname = str(selected_section.position).zfill(2) + '-' + directory_name(selected_section.name)
+        section_dirname = "%02d-%s" % (selected_section.position, selected_section.name)
         target_dir = os.path.join(args.output_dir, coursename, section_dirname)
         counter = 0
         for subsection in selected_section.subsections:
             units = all_units.get(subsection.url, [])
             for unit in units:
                 counter += 1
-                filename_prefix = str(counter).zfill(2)
+                filename_prefix = "%02d" % counter
                 if unit.video_youtube_url is not None:
                     filename = filename_prefix + "-%(title)s.%(ext)s"
                     fullname = os.path.join(target_dir, filename)
 
                     cmd = BASE_EXTERNAL_CMD + ['-o', fullname, '-f',
-                                               video_format_option,
-                                               subtitles_option]
+                                               video_format_option]
+                    if args.subtitles:
+                        cmd.append('--all-subs')
```

Comment:
Does this 'empty string' really breaks the script, weird, I take it since it is a fix, however I am pro the old form, the less 'if' in that gigantic loop the best.
btw. I have already refactored that function in an upcoming PR, I will rebase it to include your changes.